src/dvb.c

```diff
@@ -1733,13 +1733,22 @@ void get_signal(adapter *ad, int *status, uint32_t *ber, uint16_t *strength,
         }
     }
 
-    if (opts.enigma) {
-        // In Enigma2 STRENGTH:[0..100] and SNR:dB*1000
-        int64_t strengthd = 0, snrd = 0, init_snr = 0;
-        float tf;
+    if (ad->force_strength_mode == 1) {
```

Comment:
As with the previous code, this is not the right place to change the values returned by the hardware.
This method should only get the values from hardware.

I propose adding this code to a separate method and calling it from here: https://github.com/catalinii/minisatip/blob/master/src/dvb.c#L1875

Reply:
Done!